# 1.1 Problema Lineal de Producción de Helados 

### 1.2 Objetivos

#### 1.2.1 Objetivo general.
Determinar el plan semanal de producción de los diferentes tipos de paletas que conforman la “Gama Gourmet”, con el objetivo de maximizar  beneficios. 

#### 1.2.2 Objetivos específicos
* Cumplir con la demanda semanal de la empresa. 
* Maximizar la ganancia de la empresa.
* Determinar la cantidad de paletas a producir de cada sabor.

### 1.3 Modelo que representa el problema

Nuestra empresa de origen valenciano afincada en Sevilla desde la década de los años 70, se dedica a la elaboración de helados artesanos. Después de estos años de grandes progresos en su negocio, desea abrir mercado para poder enfrentarse a la situación actual.
Esta ampliación tiene como objetivo introducir sus productos en el sector de la hostelería, mediante la propuesta de una gama de helados que podemos considerar “Gourmet”. A continuación detallaremos dicha gama.

Creada por el gran Filippo Zampieron está compuesta por cinco tipos de paletas artesanales: 
1. Paletas de menta
2. Paletas de chocolate
3. Paletas de yogurt y melocotón 
4. Paletas de almendras  
5. Paletas “Fiordilatte”.

Aunque la elaboración de todas las paletas difieren en diversos aspectos, ya sea en la composición de la base, cobertura o en las proporciones de cada componente, hay un producto común en todas ellas; “*Jarabe Base*” ya que sin este no sería posible la fabricación de la base de las paletas.

Este Jarabe, está compuesto por: 
* Agua: 655 gr
* Azúcar de caña : 180 gr
* Dextosa: 35 gr
* Glucosa: 130 gr
    
A continuación detallamos el proceso de elaboración y las cantidades utilizadas
para la fabricación de un kilo de cada tipo de paletas.


#### <font color="darkblue">Paletas de menta
<img style="float: right; margin: 0px 0px 15px 15px;" src="https://likeavegan.com.au/wp-content/uploads/2014/10/chocmintpop.jpg" width="200px" height="75px" />

La fabricación de este producto comienza con la elaboración de la base. Para ello se utilizan $550 gr$ del jarabe, seguido de unas gotas de esencia de menta ($10 gotas$) y posteriormente añadiendo unos $450 gr$ de leche fresca entera.
Una vez que se ha mezclado la base y se ha dejado reposar para conseguir una textura idónea se procede a la elaboración de su cobertura.
Está compuesta por unos $800 gr$ chocolate y $200 gr$ de manteca de cacao.

#### <font color="darkblue">Paletas de Chocolate
<img style="float: left; margin: 10px 10px 10px 10px;" src="https://s23991.pcdn.co/wp-content/uploads/2011/08/mexican-chocolate-pops-recipe.jpg" width="150px" height="50px" />
La base de estas está compuesta por: $500 gr$ de jarabe, $440 gr$ de leche entera fresca unos $25 gr$ de azúcar invertido (una combinación de glucosa y fructosa) y por último, 35 gr de cacao.
La cobertura al igual que el producto anterior está compuesta por: $800 gr$ de chocolate y $200 gr$ de manteca de cacao.

#### <font color="darkblue">Paletas de Yogurt y Melocotón
<img style="float: right; margin: 10px 10px 10px 10px;" src="https://cookieandkate.com/images/2015/08/peach-popsicles-recipe.jpg" width="150px" height="50px" />
Con una base compuesta por: $430 gr$ de jarabe, $300 gr$ de yogurt desnatado, $20 gr$ de azúcar invertido y $250 gr$ de melocotón batido.
Su cobertura es una dulce combinación de $500 gr$ de chocolate y $500 gr$ de nata.

#### <font color="darkblue">Paletas de almendra
<img style="float: left; margin: 10px 10px 10px 10px;" src="https://www.thelittleepicurean.com/wp-content/uploads/2015/01/banana-coconut-ice-pops.jpg" width="200px" height="50px" />
Base elaborada por: $400 gr$ de jarabe, $495 gr$ de leche fresca entera, $25 gr$ de azúcar invertido.
La cobertura está elaborada por $800 gr$ de chocolate, $200 gr$ de manteca de cacao y $80 gr$ de pasta de almendras.

#### <font color="darkblue">Paletas “Fiordilatte”
<img style="float: right; margin: 10px 10px 15px 15px;" src="https://justbakedbyme.files.wordpress.com/2012/07/vanilla-bean-custard-pops-7.jpg?w=768&h=559" width="200px" height="50px" />
Su elaboración comienza con la base compuesta por $510 gr$ de jarabe, $510 gr$ de leche fresca entera, $250 gr$ de nata, $200 gr$ de azúcar invertido.
Una vez que la base se haya mezclado y adoptado la textura deseada, se le inyecta un relleno compuesto por: $550 gr$ de nata y $500 gr$ de chocolate.
Finalmente, esperado el tiempo necesario para que el relleno se adapte a la base, se le añade una cobertura de $800 gr$ de chocolate y $200 gr$ de manteca de cacao.

**Se ha realizado un estudio de mercado y de producción, que nos proporcionan los siguientes datos:**

#### <font color="blue">*Beneficio esperado por cada kilo de las diferentes paletas:*

* Paletas de menta: $23$ pesos el kilo
* Paletas de chocolate: $22.5$ pesos el kilo
* Paletas de yogurt y melocotón: $21$ pesos el kilo  
* Paletas de almendras: $20.5$ pesos el kilo
* Paletas “ Fiordilatte”: $21$ pesos el kilo

#### <font color="blue">*Disponibilidad semanal de las siguientes materias primas :*

* Jarabe Base: $20,5$ kg
* Leche Fresca entera: $13$ kg
* Yogurt desnatado: $5$ kg
* Nata: $8,5$ kg
* Azúcar invertido: $1,3$ kg
* Chocolate: $27$ kg
* Manteca de cacao: $5$ kg
* Esencia de menta: $2$ frasco de $90 ml$, cada frasco proporciona $75$ gotas.
* Cacao: $0,28$ kg ( $2$ bolsas de $140$ gr cada uno)
* Melocotón batido: $4$ kg
* Pasta de almendras: $0,8$ kg ( $2$ bolsas de $400$ gr cada una)


#### <font color="blue">*Demanda esperada semanalmente para cada tipo de paleta:*
* Demanda de paletas de menta y paletas de chocolate: $10$ kilos 
* Demanda de paletas de yogurt y paletas de almendras: $10$ kilos 
* No se ha estimado demanda alguna de paletas Fiordilatte.


#### <font color="blue">*Variables de decisión:*
* $x_1$ kilos a fabricar semanalmente de paletas de menta
* $x_2$ kilos a fabricar semanalmente de paletas de chocolate
* $x_3$ kilos a fabricar semanalmente de paletas de yogur y melocotón
* $x_4$ kilos a fabricar semanalmente de paletas de almendras
* $x_5$ kilos a fabricar semanalmente de paletas fiordilatte

#### <font color="blue">*Restricciones*

Limitación de Jarabe Base: $20,5$ kilos

$550x_1+500x_2+430x_3+400x_4+510x_5+\leq20500$

Limitación de Leche Fresca Entera: $13$ kilos

$450x_1+440x_2+495x_4+510x_5\leq13000$

Limitación de Yogurt desnatado: $5$ kilos 

$300x_3\leq5000$

Limitación de Nata: $8,5$ kilos 

$500x_3+550x_5\leq8500$

Limitación de Azúcar invertido: $1,3$ kilo 

$25x_2+20x_3+25x_4+200x_5\leq1300$

Limitación de Chocolate: $27$ kilos

$800x_1+800x_2+500x_3+800x_4+1300x_5\leq27000$

Limitación de Manteca de cacao: 5 kilos

$200x_1+200x_2+200x_4+200x_5\leq5000$

Limitación de Esencia de menta: $150$ gotas

$10x_1\leq150$

Limitación de Cacao: 0,28 kg 

$35x_2\leq280$

Limitación de Melocotón Batido: 4 kilos

$250x_3\leq4000$

Limitación de Pasta de Almendras: 0.8 kilo

$80x_4\leq800$

Restricciones con respecto a la demanda:

$x_1+x_2\geq10$ 

$x_3+x_4\geq10$

Función Objetivo: Beneficio

$23x_1+22.5x_2 21x_3 20.5x_4 21x_5 $

### 1.4 Solución del problema de optimización.

Sintetizando las restricciones y nuestra función a optimizar obtenemos lo siguiente:

Max  $23x_1+22.5x_2+21x_3+20.5x_4+21x_5$ 

s.a. 

$550x_1+500x_2+430x_3+400x_4+510x_5\leq20500$

$450x_1+440x_2+495x_4+510x_5\leq13000$ 
     
$300x_3\leq5000$
     
$500x_3+550x5\leq8500$
     
$25x_2+20x_3+25x_4+200x_5\leq1300$
     
$800x_1+800x_2+500x_3+800x_4+1300x_5\leq27000$
     
$200x_1+200x_2+200x_4+200x_5\leq5000$
     
$10x_1\leq150$
     
$35x_2\leq280$
     
$250x_3\leq4000$
     
$80x_4\leq800$
     
$x_1+x_2\geq10$ 
     
$x_3+x_4\geq10$ 
     
$x_1,x_2,x_3,x_4,x_5\geq0$

In [2]:
import numpy as np
import scipy.optimize as opt
c = -np.array([23, 22.5, 21, 20.5, 21])
A = np.array([[550, 500, 430, 400, 510],
              [450, 440, 0, 495, 510],
              [0, 0, 300, 0, 0],
              [0, 0, 500, 0, 550],
              [0, 25, 20, 25, 200],
              [800, 800, 500, 800, 1300],
              [200, 200, 0, 200, 200],
              [10, 0, 0, 0, 0],
              [0, 35, 0, 0, 0],
              [0, 0, 250, 0, 0],
              [0, 0, 0, 80, 0],
              [-1, -1, 0, 0, 0],
              [0, 0, -1, -1, 0]])
b = np.array([20500, 13000, 5000, 8500, 1300, 27000, 5000, 150, 280, 4000, 800, -10, -10])

In [3]:
utilidad = opt.linprog(c, A_ub=A, b_ub=b)
utilidad

     fun: -876.375
 message: 'Optimization terminated successfully.'
     nit: 8
   slack: array([1070.  , 2358.75,  200.  ,  500.  ,  761.25,    0.  ,  250.  ,
          0.  ,    0.  ,    0.  ,  740.  ,   13.  ,    6.75])
  status: 0
 success: True
       x: array([15.  ,  8.  , 16.  ,  0.75,  0.  ])

### 1.5 Visualización de la solución del problema 

Una vez que hemos descrito el proceso de producción de cada producto, hemos sintetizado toda la información en el siguiente cuadro para ver de una forma más clara, los requisitos de materias primas por tipo de paleta

|Jarabe|Leche entera fresca|Yogurt desnatado|Nata|Azúcar inv.|Chocolate|Manteca cacao|Esencia menta|Cacao|Melocotón|Pasta de almendras
:----|----
P. Menta|550gr|450gr||||800gr|200gr|10gr
P. Chocolate|500|440|||25|800|200||35
P. Yogurt y Melocotón|430||300|500|20|500||||250
P. Almendra|400|495|||25|800|200||||80
P. Fiordilatte|510|510||550|200|1300|200

Teniendo una demanda:


$P. Menta+P. Chocolate\geq10$ 
     
$P. Yogurt y melocotón + P. Almendra\geq10$ 

In [4]:
resultado= utilidad.x
resultado

array([15.  ,  8.  , 16.  ,  0.75,  0.  ])

In [5]:
excedente= utilidad.slack
excedente

array([1070.  , 2358.75,  200.  ,  500.  ,  761.25,    0.  ,  250.  ,
          0.  ,    0.  ,    0.  ,  740.  ,   13.  ,    6.75])

### 1.6 Conclusiones 



Nuestra empresa va a obtener un beneficio máximo de **876,375 pesos** cuando se producen $15 k$ de paletas de menta, $8 k$ de paletas de chocolate, $16 k $ de paletas de yogurt y melocotón junto con $0.75 k $ de paletas de almendra. En esta producción óptima no se contempla la fabricación de las paletas de "Fiordilatte" y como estas no estan contempladas en nuestra demanda no afecta nuestra proyección de ventas. 

Cuando se cumple esta producción óptima vamos a agotar las siguientes materias primas:
* Chocolate
* Esencia de menta
* Cacao 
* Melocotón batido

Mientras que quedamos con lo siguiente en el almacén:
* $1070 gr$ de Jarabe Base
* $2358.75 gr$ de leche entera fresca
* $200 gr$ de yogurt desnatado
* $500 gr$ de nata
* $761.25 gr$ de azúcar invertido
* $250 gr$ de manteca de cacao
* $740 gr$ de pasta de almendras

Con respecto a la demanda, encontramos un excedente de $13 k$ de paletas de menta y chocolate, mientras que son $6.75 k$ el excedente de paletas de almendra y yogurt con melocotón. 

### 1.7 Referencias 
* https://idus.us.es/xmlui/bitstream/handle/11441/42359/conejero%20calvo%20de%20leon%20elena%20tr.pdf?sequence=1